textattack/shared/attack.py

```diff
@@ -4,29 +4,32 @@
 import random
 
 from textattack.shared import utils
-from textattack.constraints import Constraint
+from textattack.constraints import Constraint, ModificationConstraint
 from textattack.shared import TokenizedText
-from textattack.attack_results import SkippedAttackResult
+from textattack.attack_results import SkippedAttackResult, SuccessfulAttackResult, FailedAttackResult
 
 class Attack:
     """
     An attack generates adversarial examples on text. 
     
     This is an abstract class that contains main helper functionality for 
-    attacks. An attack is comprised of a search method, a goal function, and a 
-    transformation, as well as one or more linguistic constraints that 
-    successful examples must meet.
+    attacks. An attack is comprised of a search method, which makes use of
+    a goal function, a transformation, and a set of one or more linguistic 
```

Comment:
this is ambiguous language -

maybe say "a search method and corresponding goal function, a transformation, ..."